### PR TITLE
GAWB-2251: endpoint to get permissions for multiple methods in one call

### PR DESCRIPTION
**Pull request by davidangb**
in support of 1-click sharing, create a new Agora endpoint that returns the permissions for multiple methods in one call.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/test/scala/org/broadinstitute/dsde/agora/server/webservice/PermissionIntegrationSpec.scala [hunk 1]
```diff
@@ -246,4 +249,48 @@ class PermissionIntegrationSpec extends FlatSpec with RouteTest with ScalatestRo
...
+        }
+        // check second - it exists, but we don't have permissions to see it
+        {
+          val stubEntity = AgoraEntity(agoraEntity2.namespace, agoraEntity2.name, agoraEntity2.snapshotId)
+          val found = entityAclList.find(_.entity == stubEntity)
+          assert(found.isDefined)
+          assert(found.get.message.nonEmpty)
+          assert(found.get.acls.isEmpty)
+        }
+        // check third - it doesn't exist in the db
```
Comment (aednichols) [inline src/test/scala/org/broadinstitute/dsde/agora/server/webservice/PermissionIntegrationSpec.scala:289]: Would it be worthwhile sense to test a redacted entity? The behavior should be the same as for a non-existent entity, I would think.

File: src/test/scala/org/broadinstitute/dsde/agora/server/webservice/PermissionIntegrationSpec.scala [hunk 2]
```diff
@@ -246,4 +249,48 @@ class PermissionIntegrationSpec extends FlatSpec with RouteTest with ScalatestRo
...
+
+    Post(ApiUtil.Methods.withLeadingVersion + "/permissions", payload) ~>
+      methodsService.multiEntityPermissionsRoute ~>
+      check {
+        assert(status == OK)
+        val entityAclList = responseAs[Seq[EntityAccessControl]]
+        assertResult(3) {entityAclList.size}
+
+        // check first - should get permissions
+        {
```
Comment (aednichols) [inline src/test/scala/org/broadinstitute/dsde/agora/server/webservice/PermissionIntegrationSpec.scala]: I like the standalone scope braces, I think this is the first time I've seen someone doing this in practice

**Review by aednichols — COMMENTED** [review]
Looks good, the response format is intuitive.

Some questions/issues besides inline comments:
- If you have an unshared method and share it with one reader, the reader's role appears as expected. There also appears an ACL entry for `public` with no roles. Why?
- Omitting one of the required fields in `configurations`/`methods` triggers error that's not very helpful:
```
{
  "code": 500,
  "message": "None.get"
}
```

File: src/test/scala/org/broadinstitute/dsde/agora/server/webservice/PermissionIntegrationSpec.scala (hunk 1 above)
Comment (davidangb) [reply src/test/scala/org/broadinstitute/dsde/agora/server/webservice/PermissionIntegrationSpec.scala:289]: I've added a redacted entity to the test. Under the covers, redaction removes all permissions from the entity, so it's actually the same as the second test - it exists in the db, but we're not allowed to see it.

**Comment by davidangb** [comment]
I've added better validation for when the user fails to supply a name/namespace/snapshotId.

The "public" permission is a weird quirk. I've seen that before, too. It's certainly not a result of this PR (which is read-only to the db) ... something we might want to fix at some point, even though it's harmless.

Back to you @aednichols for re-review!

**Review by aednichols — APPROVED** [review]
👍

**Pull request merged**